### PR TITLE
feat:Fetch Users flow implemented

### PR DESCRIPTION
**Pull request by AhmedHashir10**

### CODE REVIEW

File: apps/sentry/src/connectors/users.ts [hunk 1]
```diff
@@ -1,27 +1,69 @@
...
-  id: string;
-  username: string;
-  email: string;
+import { env } from '@/env';
+import { SentryError } from './commons/error';
+
+export type SentryUser = {
+  role: string;
+  user: {
+    id: string;
```
Comment (awan7676) [inline apps/sentry/src/connectors/users.ts:7]: We have also access to the field "has2fa" in api. we need to access that too.

File: apps/sentry/src/app/install/action.tsx [hunk 2]
```diff
@@ -15,6 +16,7 @@ export type FormState = {
   redirectUrl?: string;
   errors?: {
     token?: string[] | undefined;
+    sourceOrganizationId?: string[] | undefined;
```
Comment (awan7676) [inline apps/sentry/src/app/install/action.tsx]: i think we are getting organization slug an not the organization id. Can you confirm this?

**Review by awan7676 — CHANGES_REQUESTED** [review]

File: apps/sentry/src/connectors/users.ts [hunk 3]
```diff
@@ -1,27 +1,69 @@
...
+  token: string,
+  sourceOrganizationId: string,
+  cursor: string | null
+) => {
+  const headers: Record<string, string> = {
+    Authorization: `Bearer ${token}`,
+  };
+
+  const url = `https://sentry.io/api/0/organizations/${sourceOrganizationId}/members/?per_page=${env.USERS_SYNC_BATCH_SIZE}${
+    cursor ? `&cursor=${cursor}` : ''
```
Comment (awan7676) [inline apps/sentry/src/connectors/users.ts]: const url = new URL('const url = new URL('https://api.vercel.com/v2/teams/${teamId}/members'
  if (cursor !== null){
    url.searchParams.append('cursor', cursor)
  }')

File: apps/sentry/src/connectors/users.test.ts [hunk 4]
```diff
@@ -1,65 +1,108 @@
...
+        if (request.headers.get('Authorization') !== `Bearer ${validToken}`) {
+          return new Response(undefined, { status: 401 });
+        }
+        const url = new URL(request.url); 
+        const lastCursor = 'last-cursor';
+        const nextCursor = 'next-cursor';
+        const requestCursor = url.searchParams.get('cursor');
+
+        if (requestCursor !== lastCursor) {
+          return new Response(
```
Comment (awan7676) [inline apps/sentry/src/connectors/users.test.ts]: create a single response and return headers conditionally(? statement) by checking requestCursor !== lastCursor.

File: apps/sentry/src/connectors/users.test.ts [hunk 5]
```diff
@@ -1,65 +1,108 @@
...
 
-    test('should throws when the token is invalid', async () => {
-      await expect(getUsers('foo-bar', 0)).rejects.toBeInstanceOf(MySaasError);
-    });
+  test('should throw SentryError when token is invalid', async () => {
+    try {
+      await deleteUser("invalidToken", sourceOrganizationId, memberId);
+    } catch (error) {
+      expect(error instanceof SentryError).toBe(true);
+      expect(error.message).toEqual('Could not delete Sentry user');
```
Comment (awan7676) [inline apps/sentry/src/connectors/users.test.ts:94]: single line would be enough:
expect((error as SentryError).message).toEqual('Could not delete Sentry user');

File: apps/sentry/src/inngest/client.ts [hunk 6]
```diff
@@ -1,21 +1,33 @@
...
+   };
+   'sentry/users.delete.requested': {
+     data: {
+       id: string;
+       organisationId: string;
+       region:string;
+     };
+   };
+ }>(),
+ middleware: [sentryMiddleware],
```
Comment (awan7676) [inline apps/sentry/src/inngest/client.ts]: add rateLimitMiddleware too.

File: apps/sentry/src/inngest/functions/users/sync-users-page.ts [hunk 7]
```diff
@@ -1,96 +1,115 @@
...
-  id: user.id,
-  displayName: user.username,
-  email: user.email,
+
+const formatElbaUser = (user: SentryUser): User => ({
+  id: user.user.id,
+  displayName: user.user.name,
+  email: user.user.email,
+  role: user.role,
+  authMethod: undefined,
```
Comment (awan7676) [inline apps/sentry/src/inngest/functions/users/sync-users-page.ts]: when we access has2fa, then check user.has2fa? 'mfa' : 'password'

File: apps/sentry/src/connectors/users.ts (hunk 1 above)
Comment (AhmedHashir10) [reply apps/sentry/src/connectors/users.ts:7]: Done

File: apps/sentry/src/app/install/action.tsx (hunk 2 above)
Comment (AhmedHashir10) [reply apps/sentry/src/app/install/action.tsx]: Done

File: apps/sentry/src/connectors/users.ts (hunk 3 above)
Comment (AhmedHashir10) [reply apps/sentry/src/connectors/users.ts]: Done

File: apps/sentry/src/connectors/users.test.ts (hunk 4 above)
Comment (AhmedHashir10) [reply apps/sentry/src/connectors/users.test.ts]: Done

File: apps/sentry/src/connectors/users.test.ts (hunk 5 above)
Comment (AhmedHashir10) [reply apps/sentry/src/connectors/users.test.ts:94]: Done

File: apps/sentry/src/inngest/client.ts (hunk 6 above)
Comment (AhmedHashir10) [reply apps/sentry/src/inngest/client.ts]: Done

File: apps/sentry/src/inngest/functions/users/sync-users-page.ts (hunk 7 above)
Comment (AhmedHashir10) [reply apps/sentry/src/inngest/functions/users/sync-users-page.ts]: Done

**Pull request merged**